File: src/Homeworks/Homework1.Test/ProgramTests.cs
using System.Collections.Generic;
using Xunit;

namespace Homework1.Test
{
	public class ProgramTests
	{
		[Fact]
		public void Main_CorrectExpression_ReturnZero()
		{
			var actual = Program.Main(new[] {"1", "+", "2"});
			Assert.Equal(0, actual);
		}

		[Theory]
		[MemberData(nameof(Expressions))]
		public void Main_IncorrectExpression_ReturnZero(string[] args, int expected)
		{
			var actual = Program.Main(args);
			Assert.Equal(expected, actual);
		}

		public static IEnumerable<object[]> Expressions()
		{
			yield return new object[] {new[] {"11", "^", "231"}, 2};
			yield return new object[] {new[] {"s", "+", "2"}, 1};
		}
	}
}

File: src/Homeworks/Homework4.Tests/ProgramTests.fs
module ProgramTests

open Homework4
open Xunit

let checkProgram args expected =
    let actual = Program.main args
    Assert.Equal(expected, actual)

[<Fact>]
let ``Main_CorrectExpression_ReturnZero`` () =
    checkProgram [|"1"; "+"; "2"|] 0

let  Expressions: obj[][] = [|
    [|[|"11"; "^"; "231"|]; 2|]
    [|[|"s"; "+"; "2"|]; 1|]
|]

[<Theory>]
[<MemberData(nameof(Expressions))>]
let ``Main_IncorrectExpression_ReturnNotZero`` (args, expected) =
    checkProgram args expected
    